google/cloud/bigquery/schema.py

```diff
@@ -105,7 +105,12 @@ def __init__(
         if max_length is not _DEFAULT_VALUE:
             self._properties["maxLength"] = max_length
         self._fields = tuple(fields)
-        self._policy_tags = policy_tags
+
+        if policy_tags is None:
+            is_record = field_type is not None and field_type.upper() in _STRUCT_TYPES
+            self._policy_tags = None if is_record else PolicyTagList()
+        else:
+            self._policy_tags = policy_tags
```

Comment:
Can we move this logic out to a helper function, rather than replicating it in `from_api_repr` below?